### DISCUSSION TITLE
Akita Facade Pattern

### DISCUSSION THREAD

**Discussion by boissierflorian** [salesforce/akita#705] [General]
Hello Guys,

I would like to have your opinion about whether it's worth to implement the Facade pattern with Akita ?. 
Thanks for your answers

**Comment by NetanelBasal** [comment]
What are the benefits you see?

**Reply by boissierflorian** [reply]
Well, this came into my mind because I found an article which discuss about it but with an other state management library (https://auth0.com/blog/ngrx-facades-pros-and-cons/) and this (https://thomasburlesonia.medium.com/ngrx-facades-better-state-management-82a04b9a1e39). These concepts are not only related to ngrx but to any state management system. I would like to have your opinion about it because adding an other layer has a cost..

**Reply by NetanelBasal** [reply]
I think this extra layer is redudant. You can encapsulate everything inside the query or the store.

**Reply by boissierflorian** [reply]
Thank you for your clarification.
